### PR TITLE
Ps 5749 8.0

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
https://ps80.cd.percona.com/job/percona-server-8.0-param/235/

### CODE REVIEW

File: storage/innobase/fil/fil0crypt.cc [hunk 1]
```diff
@@ -886,7 +886,6 @@ static bool fil_crypt_start_encrypting_space(fil_space_t *space) {
                                                     // space from MK encryption
                                                     // to RK encryption
     crypt_data->encryption_rotation = Encryption::MASTER_KEY_TO_KEYRING;
-    ut_ad(space->encryption_key != NULL && space->encryption_iv != NULL);
```
Comment (robgolebiowski) [inline storage/innobase/fil/fil0crypt.cc:889]: ut_ad(std::all_of(space->encryption_key, space->encryption_key + ENCRYPTION_KEY_LEN, [](byte b) { b ==0; }) == false); ?

**Review by robgolebiowski — CHANGES_REQUESTED** [review]

File: plugin/tokudb-backup-plugin/tokudb_backup.cc [hunk 2]
```diff
@@ -474,8 +474,6 @@ static void tokudb_backup_before_stop_capt_fun(void *arg) {
       return;
     }
   }
-  // TODO : Backup lock not ported to 8.0
-  // (void)lock_binlog_for_backup(thd);
```
Comment (george-lorch) [inline plugin/tokudb-backup-plugin/tokudb_backup.cc:478]: This isn't exactly correct, we need to properly implement this in 8.0 for toku hot backup to actually work properly w.r.t. capturing binlog position.

**Review by george-lorch — APPROVED** [review]

File: plugin/tokudb-backup-plugin/tokudb_backup.cc (hunk 2 above)
Comment (laurynas-biveinis) [reply plugin/tokudb-backup-plugin/tokudb_backup.cc:478]: I will make the comment mention pfs.log_status instead

File: storage/innobase/fil/fil0crypt.cc (hunk 1 above)
Comment (laurynas-biveinis) [reply storage/innobase/fil/fil0crypt.cc:889]: Testing (minus "== false" part)

**Comment by laurynas-biveinis** [comment]
https://ps80.cd.percona.com/job/percona-server-8.0-param/247/

**Review by robgolebiowski — APPROVED** [review]

File: plugin/tokudb-backup-plugin/tokudb_backup.cc (hunk 2 above)
Comment (george-lorch) [reply plugin/tokudb-backup-plugin/tokudb_backup.cc:478]: LGTM

**Review by george-lorch — APPROVED** [review]

**Pull request merged**